### PR TITLE
Work item endopoints

### PR DESCRIPTION
**Pull request by leosoubeste**
Created Get, Put, Patch, Post, Delete methods for work item entity.

### CODE REVIEW

File: lib/api/v1/work_items.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          present :work_item, work_item, with: Api::V1::Entities::WorkItem
+        end
+
+        desc "EDIT a work_item"
+        params do
+          requires :id
+          requires :name
+          requires :description
+          requires :status
+          requires :user_id
```
Comment (falvarisqueta) [inline lib/api/v1/work_items.rb]: You can encapsulate work_item attributes in a work_item json.. and then read the attributes from there.
requires: id
requires: work_item

File: lib/api/v1/work_items.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        params do
+          requires :name
+          requires :status
+          requires :user_id
+        end
+        post do
+          user = User.find(params[:user_id])
+          work_item = WorkItem.create(name: params[:name],
+                                      description: params[:description],
+                                      status: params[:status],
```
Comment (falvarisqueta) [inline lib/api/v1/work_items.rb]: what happens if params does not have description?

File: spec/api/work_item_spec.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      Then { expect(response.status).to eq 200 }
+      And { expect(json_body['work_item']['name']).to eq 'Edit WI Test' }
+      Given(:ret_wi) { WorkItem.first }
+      Then { expect(ret_wi.name).to eq 'Edit WI Test' }
+      And { expect(ret_wi.description).to eq 'new edited desc' }
+      And { expect(ret_wi.status).to eq 'In Progress' }
+    end
+  end
+
+  describe "PATCH edit existing work item" do
```
Comment (falvarisqueta) [inline spec/api/work_item_spec.rb]: We usually dont use Patch in dispense

File: lib/api/v1/work_items.rb (hunk 2 above)
Comment (leosoubeste) [inline lib/api/v1/work_items.rb]: It will be null. It is not validated in the model, so I did it following that logic.

File: lib/api/v1/users.rb [hunk 4]
```diff
@@ -28,9 +28,7 @@ class Users < Grape::API
         end
         put do
           user = User.find(params[:id])
-          user.name = params[:name]
-          user.email = params[:email]
-          user.save
+          user.update_attributes!(name: params[:name], email: params[:email])
```
Comment (ignaciojonas) [inline lib/api/v1/users.rb]: Nice! :)

File: lib/api/v1/work_items.rb [hunk 5]
```diff
@@ -0,0 +1,74 @@
+module Api
+  module V1
+    class WorkItems < Grape::API
+      version 'v1' # path-based versioning by default
+      format :json
+
+      resource :work_items do
+        desc "Return LIST of work_items", { entity: Api::V1::Entities::WorkItem }
+        get do
+          #User.all
```
Comment (ignaciojonas) [inline lib/api/v1/work_items.rb]: This line can be removed.

File: lib/api/v1/work_items.rb (hunk 1 above)
Comment (ignaciojonas) [inline lib/api/v1/work_items.rb]: You can take a look to dispense_type in the dispense project.

Comment (leosoubeste) [inline lib/api/v1/work_items.rb]: image

**Comment by ignaciojonas** [comment]
Looks good to merge for me...
@falvarisqueta?

**Comment by falvarisqueta** [comment]
Looks good

**Pull request merged**